### PR TITLE
Add DNS provider for CloudDNS.

### PR DESCRIPTION
**Pull request by radek-sprta**
Adds support for [Clouddns](https://github.com/vshosting/clouddns). Let me know your feedback :)

Here are results from issuing a wildcard certificate: [https://pastebin.vshosting.cz/?52c614bff3e9b9dd#6JuGJegnZ2bJNLDQ1r1WHAE8x88YfHX5toCJi68Fk1rH](https://pastebin.vshosting.cz/?52c614bff3e9b9dd#6JuGJegnZ2bJNLDQ1r1WHAE8x88YfHX5toCJi68Fk1rH)

### CODE REVIEW

**Comment by ldez** [comment]
Hello, in order for a PR adding a DNS provider to be accepted, you have to:

- [ ] add a description to your PR
- [ ] be able to maintain this provider
- [ ] have a homogeneous design with the other providers
- [ ] add tests (units)
```shell
make test
```
- [ ] add tests ("live")
https://github.com/go-acme/lego/blob/c938de68f262c057542d44bcb67ef961cefcc60e/providers/dns/cloudflare/cloudflare_test.go#L125-L151

```shell
make test
```
- [ ] add a [provider descriptor](https://github.com/go-acme/lego/blob/master/providers/dns/cloudflare/cloudflare.toml)
- [ ] generate CLI help, documentation, and readme
```bash
make generate-dns
```
- [ ] be able to do: _(and put the output of this command to a comment in your PR)_
```bash
rm -rf .lego

./lego -m your@email.com --dns YOUR_PROVIDER_NAME -d *.example.com -d example.com -s https://acme-staging-v02.api.letsencrypt.org/directory run
```
Note the wildcard domain is important.
- [ ] pass the linter ([golangci-lint](https://github.com/golangci/golangci-lint#install) must be installed):
```shell
make checks
```
- [ ] do `go mod tidy`

File: providers/dns/clouddns/internal/client.go [hunk 1]
```diff
@@ -37,199 +25,145 @@ type Client struct {
...
 func (c *Client) login() error {
 	reqData := authorization{Email: c.Email, Password: c.Password}
+
 	body, err := json.Marshal(reqData)
 	if err != nil {
 		return err
 	}
 
-	req, err := http.NewRequest("POST", loginURL, bytes.NewReader(body))
+	req, err := http.NewRequest(http.MethodPost, loginURL, bytes.NewReader(body))
```
Comment (marcofranssen) [inline providers/dns/clouddns/internal/client.go:166]: why not use the `doAPIRequest` method here?

**Comment by ldez** [comment]
ping @radek-sprta

**Comment by radek-sprta** [comment]
Sorry for the lack of communication. I had some time off, so I'll take a proper look at this on Monday.

**Comment by ldez** [comment]
@radek-sprta any news?

**Comment by radek-sprta** [comment]
> @radek-sprta any news?

I have added the results of issuing a wildcard certificate in the original post.

I have also subscribed to the repository. When an issue comes up with a DNS plugin, do you assign them to individual plugin maintainers when triaging, or is it up to them to watch the issues? I am asking in case I want to add somebody else from our team as a maintainer in the future.

**Pull request merged**